### PR TITLE
feat: upgrade to django 4.2

### PR DESCRIPTION
**Pull request by irtazaakram**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.5, 3.6, 3.7, 3.8, 3.9, pypy-3.7]
+        python-version: [3.8, 3.9, "3.10"]
```
Comment (UsamaSadiq) [inline .github/workflows/build.yml]: Let's only keep the mentioned version and add only support for `Django` for now.

File: setup.py [hunk 2]
```diff
@@ -37,11 +37,9 @@ def read(*parts):
         'Topic :: Software Development :: Libraries :: Python Modules',
         'Framework :: Django',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.5',
-        'Programming Language :: Python :: 3.6',
-        'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
```
Comment (UsamaSadiq) [inline setup.py:42]: Let's skip adding new Python version.

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{35,36,37,38,39}-django22, py{36,37,38,39}-{django30,django31,django32}, pypy37-{django30,django31,django32}, py{38,39}-{djangomain}, lint, docs
+envlist = py{38,39}-django22, py{38,39}-{django30,django31,django32}, py{38,39,310}-django{40,41,42}, py{310}-{djangomain}, lint, docs
```
Comment (UsamaSadiq) [inline tox.ini]: We can remove `Django 2.2` as well since its endoflife has passed as well.

**Review by UsamaSadiq — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (irtazaakram) [reply .github/workflows/build.yml]: we would need to drop Django main from tests as 3.10 is required for it.

**Pull request closed without merging**